From 55e244b6c8655f2d64d1a55c8808c30110b15c2e Mon Sep 17 00:00:00 2001
From: jt1134 <jt1134@gmail.com>
Date: Mon, 20 Jan 2014 19:45:13 +0200
Subject: [PATCH 024/172] Camera: separate settings for color effects

Author: jt1134 <jt1134@gmail.com>
Date:   Mon Jan 20 19:45:13 2014 +0200

    Camera: separate settings for color effects

    Store color effect settings for the camera and
    video camera in their own separate preferences.

    Change-Id: I8faa59b8663154737910b338906d27f9736d5ad9

Author: Michael Bestas <mikeioannina@gmail.com>
Date:   Tue Aug 26 06:11:21 2014 +0300

    Snap: Fix filtering of unsupported camcorder color effects

    Change-Id: I365deeb17995175db873f6799cf0cde43ee61e96

Change-Id: I7ee41923f8a7313eee06d5e0bf7e29a365d0af86
---
 res/xml/video_preferences.xml              | 2 +-
 src/com/android/camera/CameraSettings.java | 7 +++++++
 src/com/android/camera/VideoMenu.java      | 4 ++--
 src/com/android/camera/VideoModule.java    | 2 +-
 4 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/res/xml/video_preferences.xml b/res/xml/video_preferences.xml
index f2e2e9e15..9a42a9b5f 100644
--- a/res/xml/video_preferences.xml
+++ b/res/xml/video_preferences.xml
@@ -126,7 +126,7 @@
             camera:largeIcons="@array/pref_filter_mode_icons"
             camera:entryValues="@array/pref_camera_filter_mode_entryvalues" />
     <IconListPreference
-            camera:key="pref_camera_coloreffect_key"
+            camera:key="pref_camera_video_coloreffect_key"
             camera:defaultValue="@string/pref_camera_coloreffect_default"
             camera:title="@string/pref_camera_coloreffect_title"
             camera:entries="@array/pref_camera_coloreffect_entries"
diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 446caa0c1..d019c4a3c 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -89,6 +89,7 @@ public class CameraSettings {
     public static final String KEY_CAMERA_SAVEPATH = "pref_camera_savepath_key";
     public static final String KEY_FILTER_MODE = "pref_camera_filter_mode_key";
     public static final String KEY_COLOR_EFFECT = "pref_camera_coloreffect_key";
+    public static final String KEY_VIDEOCAMERA_COLOR_EFFECT = "pref_camera_video_coloreffect_key";
     public static final String KEY_FACE_DETECTION = "pref_camera_facedetection_key";
     public static final String KEY_TOUCH_AF_AEC = "pref_camera_touchafaec_key";
     public static final String KEY_SELECTABLE_ZONE_AF = "pref_camera_selectablezoneaf_key";
@@ -734,6 +735,7 @@ public class CameraSettings {
         ListPreference powerMode = group.findPreference(KEY_POWER_MODE);
         ListPreference zsl = group.findPreference(KEY_ZSL);
         ListPreference colorEffect = group.findPreference(KEY_COLOR_EFFECT);
+        ListPreference camcorderColorEffect = group.findPreference(KEY_VIDEOCAMERA_COLOR_EFFECT);
         ListPreference faceDetection = group.findPreference(KEY_FACE_DETECTION);
         ListPreference touchAfAec = group.findPreference(KEY_TOUCH_AF_AEC);
         ListPreference selectableZoneAf = group.findPreference(KEY_SELECTABLE_ZONE_AF);
@@ -877,6 +879,11 @@ public class CameraSettings {
                     colorEffect, mParameters.getSupportedColorEffects());
         }
 
+        if (camcorderColorEffect != null) {
+            filterUnsupportedOptions(group,
+                    camcorderColorEffect, mParameters.getSupportedColorEffects());
+        }
+
         if (aeBracketing != null) {
             filterUnsupportedOptions(group,
                      aeBracketing, getSupportedAEBracketingModes(mParameters));
diff --git a/src/com/android/camera/VideoMenu.java b/src/com/android/camera/VideoMenu.java
index 330fa2ace..e891ce7af 100644
--- a/src/com/android/camera/VideoMenu.java
+++ b/src/com/android/camera/VideoMenu.java
@@ -508,7 +508,7 @@ public class VideoMenu extends MenuController
     public void initFilterModeButton(View button) {
         button.setVisibility(View.INVISIBLE);
         final IconListPreference pref = (IconListPreference) mPreferenceGroup
-                .findPreference(CameraSettings.KEY_COLOR_EFFECT);
+                .findPreference(CameraSettings.KEY_VIDEOCAMERA_COLOR_EFFECT);
         if (pref == null || pref.getValue() == null)
             return;
 
@@ -536,7 +536,7 @@ public class VideoMenu extends MenuController
 
     public void addFilterMode() {
         final IconListPreference pref = (IconListPreference) mPreferenceGroup
-                .findPreference(CameraSettings.KEY_COLOR_EFFECT);
+                .findPreference(CameraSettings.KEY_VIDEOCAMERA_COLOR_EFFECT);
         if (pref == null)
             return;
 
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 5734d174a..3d190e520 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -2334,7 +2334,7 @@ public class VideoModule implements CameraModule,
 
         setZoomMenuValue();
         String colorEffect = mPreferences.getString(
-            CameraSettings.KEY_COLOR_EFFECT,
+            CameraSettings.KEY_VIDEOCAMERA_COLOR_EFFECT,
             mActivity.getString(R.string.pref_camera_coloreffect_default));
         Log.v(TAG, "Color effect value =" + colorEffect);
         if (isSupported(colorEffect, mParameters.getSupportedColorEffects())) {
-- 
2.17.1

